extension/src/popup/views/AccountHistory/index.tsx

```diff
@@ -141,86 +143,75 @@ export const AccountHistory = () => {
       return segments;
     };
 
-    const fetchAccountHistory = async () => {
-      try {
-        const operations = await getAccountHistory(publicKey, networkDetails);
-        setHistorySegments(createSegments(operations));
-      } catch (e) {
-        console.error(e);
-      }
-    };
-
-    const getData = async () => {
-      setIsLoading(true);
-      await fetchAccountHistory();
-      setIsLoading(false);
-    };
-
-    getData();
-  }, [publicKey, networkDetails, dispatch, isHideDustEnabled]);
-
-  return isDetailViewShowing ? (
-    <TransactionDetail {...detailViewProps} />
-  ) : (
-    <>
-      <View.Content>
-        <div className="AccountHistory" data-testid="AccountHistory">
-          {isLoading ? (
-            <div className="AccountHistory__loader">
-              <Loader size="2rem" />
+    if (getHistoryState.state === RequestState.SUCCESS) {
+      setHistorySegments(createSegments(getHistoryState.data));
+    }
+  }, [
+    getHistoryState.state,
+    getHistoryState.data,
+    publicKey,
+    networkDetails,
+    isHideDustEnabled,
+  ]);
+
+  const showLoader =
```

Comment:
NIT: let's call this `isLoaderShowing` to follow code convention

Reply:
yup, renamed in 68003d30e1193bec7f21757fb3cc7afaee619967